processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryRunnerFactory.java

```diff
@@ -82,15 +83,23 @@ public SegmentMetadataQueryRunnerFactory(
         SegmentMetadataQuery query = (SegmentMetadataQuery) inQ;
 
         final QueryableIndex index = segment.asQueryableIndex();
+
         final Map<String, ColumnAnalysis> analyzedColumns;
         long totalSize = 0;
+        int numRows = 0;
```

Comment:
minor nit, but you could make this a final int without initial value, to ensure it always gets set in the if statement.